third_party/amd/lib/TritonAMDGPUTransforms/AccelerateAMDMatmul.cpp

```diff
@@ -1070,6 +1070,12 @@ class ScaledBlockedToScaledMFMAF8F6F4 final
     int vecA = deduceTilesPerWarp(aScale, 0, mDim, warpsPerTile, &tilesA);
     int vecB = deduceTilesPerWarp(bScale, 1, mDim, warpsPerTile, &tilesB);
     tilesPerWarp = vecA > vecB ? tilesA : tilesB;
+    // the chosen tilesPerWarp should not cross block boundary
+    unsigned m = aScale.getType().getShape()[0];
+    unsigned n = bScale.getType().getShape()[0];
+    tilesPerWarp[0] = std::min(tilesPerWarp[0], m / mDim);
+    tilesPerWarp[1] = std::min(tilesPerWarp[1], n / nDim);
```

Comment:
I think we should add this into the `deduceTilesPerWarp` utility function to skip not suitable choices so that we don't need to duplicate this logic in all callsites.

Reply:
will do.

Reply:
done.